lib/src/Base/Optim/HiGHS.cxx

```diff
@@ -0,0 +1,235 @@
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include "openturns/HiGHS.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+#include "openturns/OTconfig.hxx"
+
+#ifdef OPENTURNS_HAVE_HIGHS
+#include "Highs.h"
+#endif
+
+#include <chrono>
+
+BEGIN_NAMESPACE_OPENTURNS
+
+CLASSNAMEINIT(HiGHS)
+
+static const Factory<HiGHS> Factory_HiGHS;
+
+/* Constructor with no parameters */
+HiGHS::HiGHS()
+  : OptimizationAlgorithmImplementation()
+{
+  // Nothing to do here
+}
+
+
+/* Constructor that sets starting sample */
+HiGHS::HiGHS(const OptimizationProblem & problem)
+  : OptimizationAlgorithmImplementation(problem)
+
+{
+}
+
+
+/* Check whether this problem can be solved by this solver */
+// Inherited method: never used by HiGHS because solver_ calls its own checkProblem method.
+void HiGHS::checkProblem(const OptimizationProblem & problem) const
+{
+  // Cannot solve multi-objective problems
+  if (problem.hasMultipleObjective()) throw InvalidArgumentException(HERE) << "Bonmin does not support multi-objective optimization";
+
+  // No LeastSquaresProblem / NearestPointProblem
+  if (problem.hasResidualFunction() || problem.hasLevelFunction())
+    throw InvalidArgumentException(HERE) << "HiGHS does not support least squares or nearest point problems";
+
+  if (problem.hasEqualityConstraint())
+    throw InvalidArgumentException(HERE) << "HiGHS does not support equality constraints";
+}
+
+
+void HiGHS::run()
+{
+#ifdef OPENTURNS_HAVE_HIGHS
+  const UnsignedInteger problemDimension = getProblem().getDimension();
+  if (problemDimension == 0) throw InvalidArgumentException(HERE) << "No problem has been set.";
+  result_ = OptimizationResult(getProblem());
+  std::chrono::steady_clock::time_point t0 = std::chrono::steady_clock::now();
+  const UnsignedInteger initialCallsNumber = getProblem().getObjective().getCallsNumber();
+
+  HighsModel model;
+  model.lp_.num_col_ = problemDimension;
+  model.lp_.num_row_ = getProblem().getInequalityConstraint().getOutputDimension();
+
+  // objective function
+  const Point origin(problemDimension, 0.0);
+  model.lp_.sense_ = getProblem().isMinimization() ? ObjSense::kMinimize : ObjSense::kMaximize;
+  const Scalar f0 = getProblem().getObjective()(origin)[0];
+  model.lp_.offset_ = f0;
+  const Matrix grad0(getProblem().getObjective().gradient((origin)));
```

Comment:
No check on the linearity of the objective function? You could use the isLinear() method here